src/shared-components/icon/index.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import iconFilePath from '../../svgs/icons.svg';
+import { COLORS } from '../../constants';
+
+const propTypes = {
+  iconName: PropTypes.string.isRequired,
+  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  rotate: PropTypes.number,
+  fill: PropTypes.string,
+  inline: PropTypes.bool,
+  className: PropTypes.string,
+  duskId: PropTypes.string,
+};
+
+const defaultProps = {
+  className: '',
+  fill: COLORS.primary,
+  height: 15,
+  inline: false,
+  rotate: 0,
+  width: 15,
+};
+
+const Icon = ({
+  className,
+  fill,
+  height,
+  iconName,
+  inline,
+  rotate,
+  width,
+  duskId,
+  ...rest
+}) => {
+  // eslint-disable-next-line no-undef
+  const icon = window.__SVG_INLINE__
+    ? `#${iconName}`
+    : `${iconFilePath}#${iconName}`;
+  return (
+    <svg
+      width={width}
+      height={height}
+      fill={fill}
+      data-dusk-id={duskId}
+      css={`
+        display: ${inline ? 'inline-block' : 'block'};
+        transform: rotate(${rotate}deg);
+        color: ${fill};
+        transition: color 350ms;
```

Comment:
can we take this out of the component library?

Reply:
are you talking about the color transition? or this entire CSS block?

Reply:
color transition, but maybe we actually do need it? ask ben, up to you